### PR TITLE
tooling: Tweak `tools.base.utils.extract` to work with mulitple tarballs (+…

### PR DESCRIPTION
**Pull request by phlax**
…cleanups)

Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: tooling: Tweak `utils.base.extract` to work with mulitple tarballs (+cleanups)
Additional Description: 
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/base/utils.py [hunk 1]
```diff
@@ -74,14 +79,19 @@ def buffered(
...
-    with tarfile.open(tarball) as tarfiles:
-        tarfiles.extractall(path=path)
-        return path
+def extract(path: Union[pathlib.Path, str], *tarballs: Union[pathlib.Path,
+                                                             str]) -> Union[pathlib.Path, str]:
+    if not tarballs:
+        raise ExtractError(f"No tarballs specified for extraction to {path}")
+    openers = nested(*tuple(tarfile.open(tarball) for tarball in tarballs))
+    with openers as tarfiles:
+        for _tarfile in tarfiles:
```
Comment (htuch) [inline tools/base/utils.py:88]: Nit: dislike `_` prefix for locals, it confuses with private object variables.

File: tools/base/tests/test_utils.py [hunk 2]
```diff
@@ -119,39 +120,70 @@ def test_util_coverage_with_data_file(patches):
...
         prefix="tools.base.utils")
 
-    with patched as (m_tmp, m_open):
-        assert utils.extract("TARBALL", "PATH") == "PATH"
+    with patched as (m_nested, m_open):
+        _extractions = [MagicMock(), MagicMock()]
+        m_nested.return_value.__enter__.return_value = _extractions
+
+        if not tarballs:
+
```
Comment (htuch) [inline tools/base/tests/test_utils.py:138]: Tiny nit: lot of gratuitous blank lines here. Prefer compact spacing.

**Review by htuch — CHANGES_REQUESTED** [review]
LGTM modulo nits.
/wait

File: tools/base/tests/test_utils.py (hunk 2 above)
Comment (phlax) [reply tools/base/tests/test_utils.py:138]: i tend to give `if` and `with` a line above but ill remove the the line below...

Comment (phlax) [reply tools/base/tests/test_utils.py:138]: hmm actually i would prefer to keep - i think with contexts its better to have a clear line above, they are kinda almost like closure functions or something, so i think better to be explicit

File: tools/base/utils.py (hunk 1 above)
Comment (phlax) [reply tools/base/utils.py]: updated

File: tools/base/tests/test_utils.py (hunk 2 above)
Comment (phlax) [reply tools/base/tests/test_utils.py:140]: i reduced it a bit by swapping the conditional around

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**